### PR TITLE
Streaming tests added

### PR DESCRIPTION
**Pull request by Adir111**
Related to this PR: https://github.com/kube-HPC/hkube/pull/2009
which is related to this issue: https://github.com/kube-HPC/hkube/issues/2002
Added end to end tests for new streaming scaling logic.
Warning - Do not merge until the PR is changed.

### CODE REVIEW

File: tests/streaming_tests.js [hunk 1]
```diff
@@ -215,216 +113,207 @@ describe('streaming pipeline test', () => {
...
             const res = await runRaw(streamSimple);
             const { jobId } = res.body;
-            await waitForStatus(jobId, 'sen-1', 'active', 60000, 2000);
-            console.log("sen-1 is activce")
-            await delay(140);
-            await waitForStatus(jobId, 'sen-out-1', 'active', 60000, 2000);
-            console.log("sen-out-1 is activce")
-            await delay(5);
-            let required = await getRequiredPods(jobId, 'sen-1', 'sen-out-1');
+            await waitForStatus(jobId, statefulNodeName, 'active', 60 * 1000, 2 * 1000);
```
Comment (golanha) [inline tests/streaming_tests.js:125]: To be more aligned with the time of the statefull sender rates. better to put the delay after the statefull is active. Then you don't wait both full waiting time for statelss plus delay.

File: tests/streaming_tests.js
Comment (golanha) [inline tests/streaming_tests.js:1]: I would add a different pipeline such with 2 differnt previous nodes to the statless with 2 differnt rates.

File: tests/streaming_tests.js [hunk 2]
```diff
@@ -59,103 +61,12 @@ describe('streaming pipeline test', () => {
...
 
     describe("time tests", () => {
-
-        it.only("run simple stream", async () => {
-            await createAlg(start);
+        it("should satisfy the request rate with the given rate, with enough nodes.", async () => {
             await createAlg(statefull);
-            algList.push(start.name);
             algList.push(statefull.name);
             try {
```
[the remark this answers is not in the document]
Comment (Adir111) [reply tests/streaming_tests.js:71]: done

File: tests/streaming_tests.js [hunk 3]
```diff
@@ -215,216 +113,207 @@ describe('streaming pipeline test', () => {
...
+            catch (e) {
+                e.printSackTrace();
+            }
 
-//         for (const [i, jj] of jjjj.b.entries()) {
-//             console.log(i)
-//             console.log(jj)
-//         }
+            streamSimple.flowInput = createFlowInput({
+                processTime: 0.99,
```
[the remark this answers is not in the document]
Comment (Adir111) [reply tests/streaming_tests.js:221]: done

File: additionalFiles/defaults/pipelines/stream-TwoStreamingNodesWithDifferentFlows.js [hunk 4]
```diff
@@ -0,0 +1,107 @@
+const pipe = {
+    "name": "sub-ein-tst",
+    "kind": "stream",
+    "flowInput": {
+        "process_time": 0.02,
+        "flows": [
```
Comment (golanha) [inline additionalFiles/defaults/pipelines/stream-TwoStreamingNodesWithDifferentFlows.js:6]: Both flows have the save rate. The idea was to test different rates from each node

**Review by golanha — APPROVED** [review]

**Review by golanha — APPROVED** [review]

**Pull request merged**